Address safer CPP warnings in DestinationColorSpace.cpp

cdumez opened the pull request:
#### 28698492e4f86db2c0b1cbec38061176ae171229
<pre>
Address safer CPP warnings in DestinationColorSpace.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=298923">https://bugs.webkit.org/show_bug.cgi?id=298923</a>

Reviewed by Sam Weinig.

* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/platform/graphics/DestinationColorSpace.cpp:
(WebCore::operator==):
(WebCore::DestinationColorSpace::asRGB const):
(WebCore::DestinationColorSpace::asExtended const):
(WebCore::DestinationColorSpace::supportsOutput const):
(WebCore::DestinationColorSpace::usesExtendedRange const):
(WebCore::DestinationColorSpace::usesITUR_2100TF const):
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/graphics/DestinationColorSpace.h:
(WebCore::DestinationColorSpace::protectedPlatformColorSpace const):

Canonical link: <a href="https://commits.webkit.org/300115@main">https://commits.webkit.org/300115@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/18c95b2ee60ff6a766770ed3de169156b8e37495

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121298 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40995 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31653 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127734 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73379 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1647903f-9e35-48e9-a246-654149ca4040) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123174 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41697 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49574 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92141 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61301 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/883f4683-d3bd-484b-883c-c9619b6e3b66) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124250 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33293 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108699 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72817 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a277db35-1fc3-4002-bcb1-7fde69d06325) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32310 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26827 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71316 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102786 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27003 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130572 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48226 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36661 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100739 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48594 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104888 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100644 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25526 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46048 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24111 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44920 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48084 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/53797 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47555 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50902 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49238 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/platform/graphics/DestinationColorSpace.h:
```diff
@@ -63,6 +63,9 @@ class DestinationColorSpace {
     PlatformColorSpaceValue platformColorSpace() const { return m_platformColorSpace.get(); }
 #endif
+#if USE(CG)
+    RetainPtr<PlatformColorSpaceValue> protectedPlatformColorSpace() const { return platformColorSpace(); }
```
We have a type for this called `PlatformColorSpace`. Perhaps we should just have the function above (`platformColorSpace()` return the RetainPtr (e.g. `PlatformColorSpace`) type? Then we can remove the USE(SKIA) here.

cdumez replied:
I don't think we usually have getter return smart pointers unless their name starts with `protected` / `checked`. Not all callers will need a smart pointer (e.g. to call a trivial function). cc @rniwa for opinion. Maybe things are different here because ObjC?

cdumez replied:
Certainly, I can return a `PlatformColorSpace` and drop the `USE(CG)` though. That seems like a no-brainer if it works.

geoffreygaren replied:
I don't think ObjC makes this different. Raw pointer is the preferred return type if possible.

weinig replied:
That's a good rule to know!

I don't quite get it though, if even seemingly (seemingly probably being the operative word) trivial case like this require the "protected" version:

```
if (RetainPtr colorSpace = adoptCF(CGColorSpaceCreateExtended(protectedPlatformColorSpace().get())))
```

In what situations would the non-protected one be usable?

cdumez replied:
> In what situations would the non-protected one be usable?

I don't know about ObjC but for C++, functions that are inline and do not destroy objects are considered trivial (e.g. a getter) and do not require the protected variant. The "inline" requirement is only a limitation of the static analysis at the moment that we can hopefully address in the future.

cdumez committed:
Address safer CPP warnings in DestinationColorSpace.cpp
https://bugs.webkit.org/show_bug.cgi?id=298923

Reviewed by Sam Weinig.

* Source/WebCore/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/platform/graphics/DestinationColorSpace.cpp:
(WebCore::operator==):
(WebCore::DestinationColorSpace::asRGB const):
(WebCore::DestinationColorSpace::asExtended const):
(WebCore::DestinationColorSpace::supportsOutput const):
(WebCore::DestinationColorSpace::usesExtendedRange const):
(WebCore::DestinationColorSpace::usesITUR_2100TF const):
(WebCore::operator<<):
* Source/WebCore/platform/graphics/DestinationColorSpace.h:
(WebCore::DestinationColorSpace::protectedPlatformColorSpace const):

Canonical link: https://commits.webkit.org/300115@main
